configurator/configurator.go

```diff
@@ -9,18 +9,14 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
-const confFileName = "cleaner_config.yml"
+const (
+	confFileName           = "cleaner_config.yml"
+	defaultStartPath       = "."
+	defaultSizeLimit int64 = 5_000_000
```

Comment:
Numeric constants don't have to have a type assigned to them. Memory for this value will be allocated only when the constant gets assigned to some variable. The following compiles just fine:
```
const defaultSizeLimit = 123
var a uint64 = defaultSizeLimit
var b int16 = defaultSizeLimit
```

Reply:
I need `int64` explicitly  because `os.Stat` -> `Size` returns `int64` and it is incompatible with `int`. I cant compare these two types

Reply:
Check this out: https://go.dev/play/p/U5EIZ9OZkcZ . Untyped numeric constants are compatible with all numeric types with no need in explicit conversion.

Reply:
> Check this out: https://go.dev/play/p/U5EIZ9OZkcZ . Untyped numeric constants are compatible with all numeric types with no need in explicit conversion.

here is my case https://go.dev/play/p/DhVhOnBg1LW

Reply:
@efu
 oh, I understood. Your example compares two `int`
const c = 5_000_000
var i int = 50
that is why it works

Reply:
Not really.
My code works because I compare an untyped integer **constant** and an integer **variable**. This is allowed.
Your code doesn't compile because you compare **two variables** with different integer types. This isn't allowed without an explicit type casting.
```
const c = 5
var i int = 5
fmt.Println(c == i) // OK, true

var i int = 5
var i64 int64 = 5
fmt.Println(i == i64) // not OK; compilation error invalid operation: i == i64 (mismatched types int and int64)
fmt.Println(int64(i) == i64) // OK because of the explicit cast, true
fmt.Println(i == int(i64)) // OK because of the explicit cast, true
fmt.Println(int8(i) == int8(i64)) // OK because both parts of the comparison are of the same type - int8
```
This is why I recommend using an untyped integer constant on line 15 (without `int64` type specifier) - it makes it easy to use with any other variable of an integer type **without explicit casting**.